### DISCUSSION TITLE
multiple vtk files to animation ?

### DISCUSSION THREAD

**Discussion by MAT-AI-OS** [tkeskita/BVtkNodes#102] [Q&A] [answered]
hello,
I'm new in BVtk. I am trying the last days to make an animation of several *.vtk legacy files.

 I can present one mesh.  I read https://bvtknodes.readthedocs.io/en/latest/BVTKNodes.html#time-selector 
and my files are named in the exact same name except the number than represents a time step.
 
  I understood that Use Scene Time should be set to Off (how do I turn it off ? ) and Global Time Keeper Node should not be used at all in my case.
 I ran out of ideas to try. I post the nodes that work for one file. 
 
 Do you have any ideas which nodes i should use?
  I tried vtkPolyDataReader as mention, but it doesn't read the file.

![2023-08-03 13_57_29](https://github.com/tkeskita/BVtkNodes/assets/58124972/610d0724-2abf-4a76-b056-a6c8453b3a9a)
![2023-08-03 13_54_58](https://github.com/tkeskita/BVtkNodes/assets/58124972/2e5f6a4e-1e6b-4a6a-9106-d30592189e10)

Thanks in advance,
Matthew

**Comment by tkeskita** [comment, marked as answer]
Hi,

Good to know that *vtkDataSetReader* reads the data for this case (you seem to have unstructured grid data). I haven't used that reader, hope it works with *Time Selector*! You can also try *vtkUnstructuredGridReader* directly. 

You need to add *Time Selector* node between the reader and *Color Mapper* nodes. *Use Scene Time* option is located on the *Time Selector* node. When it is enabled, you can use Blender Timeline to change between time points, so keep it enabled. Hope this helps!

**Comment by MAT-AI-OS** [comment]
Thanks a lot.! It does work :) 
Time Selector selects the right filename and and it does work with vtkDataSetReader.

 I used Blender Timeline (with Scene Time enabled) and by constantly re-pressing 'Force Upstream Update' I kinda had an animation played on 3Dviewport. 
 
In Color Mapper i disabled Auto Range -so that differences between files is viewable- and enabled Generate Scalar Bar -but I get no colors in Scalar Bar.

Do you know how to make the final node auto-update for each keyframe?

I suppose after that, I can work with blender animation, camera/ rendering to make a exportable animation.
 Any suggestion is appreciated.
 
![2023-08-03 20_56_12](https://github.com/tkeskita/BVtkNodes/assets/58124972/b1a5ec1f-feb0-4471-8481-dd687fb80486)

**Reply by MAT-AI-OS** [reply]
Quite strange that in my test blender file (the one with above screenshots) I don't get animation while keyframe is changing on timeline.
But on the main file I use, where I use also Sverchok, I get animation.! I am wondering why.

I am still trying to get colors on Scalar Bar. 
If you can guess why, I will be happy to test it. Thanks in advance!

**Comment by tkeskita** [comment]
Hi,

good that *vtkDataSetReader* works! Since you're doing a somewhat involved animation, please [read carefully through the docs](https://bvtknodes.readthedocs.io/en/latest/):
- Enable "Update All Automatically" Update Mode in [Inspect](https://bvtknodes.readthedocs.io/en/latest/BVTKNodes.html#inspect) Panel. This will cause VTK pipeline to auto-update upon frame changes.
- [Scalar Bar is not working well](https://bvtknodes.readthedocs.io/en/latest/BVTKNodes.html#color-mapper), I suggest not to use it.
-  What property are you animating with keyframes? If you try to animate a BVTK node property, you need to use [Global Time Keeper node](https://bvtknodes.readthedocs.io/en/latest/BVTKNodes.html#global-time-keeper). I think normal Blender properties should be animated with keyframes without Global Time Keeper.
